train_lam.py

```diff
@@ -83,6 +83,7 @@ def lam_loss_fn(
     inputs["videos"] = inputs["videos"].astype(args.dtype) / 255.0
     model.train()
     outputs = model(inputs, training=True)
+    outputs["recon"] = outputs["recon"].astype(jnp.float32)
     gt_future_frames = inputs["videos"][:, 1:]
```

Comment:
in this case gt is not in full precision though. If you're doing this, it likely makes sense to cast the gt ones to full precision as well?